packages/api/src/routes/sensorRoute.js

```diff
@@ -32,9 +34,20 @@ router.post(
   SensorController.addSensors,
 );
 router.delete('/delete_sensor/:sensor_id', SensorController.deleteSensor);
-router.post('/add_reading/:partner_id', SensorController.addReading);
+router.post('/add_reading/:partner_id/:farm_id', validateRequest, SensorController.addReading);
 router.post('/get_readings', SensorController.getAllReadingsBySensorId);
 router.get('/sensor_readings/:farm_id/:days', SensorController.getReadingsByFarmId);
 router.post('/invalidate_readings', SensorController.invalidateReadings);
 
+function validateRequest(req, res, next) {
```

Comment:
I think to be consistent this function should be in its own file in the middleware folder.